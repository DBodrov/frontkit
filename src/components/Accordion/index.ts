import { AccordionDetailsSuccess } from './AccordionDetailsSuccess';
import { AccordionLink } from './AccordionLink';

export const Accordion = {
    Details: AccordionDetailsSuccess,
    Link: AccordionLink,
};

export { AccordionLineType } from './AccordionDetailsSuccess';
