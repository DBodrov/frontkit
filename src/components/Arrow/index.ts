export { ArrowTypes, Arrow } from './Arrow';
