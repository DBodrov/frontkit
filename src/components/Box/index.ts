export { Box, SplitType, offsetMargin, useInsideBoxClassName } from './Box';
