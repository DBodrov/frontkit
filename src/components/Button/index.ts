export { Button, StyleTypeProp } from './Button';
export { LoadingButton } from './LoadingButton';
