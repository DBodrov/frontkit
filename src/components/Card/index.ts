export { Box, SplitType } from './Box';
