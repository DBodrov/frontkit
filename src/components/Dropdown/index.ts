import { BaseDropdown } from './BaseDropdown';

export const Dropdown = BaseDropdown;
