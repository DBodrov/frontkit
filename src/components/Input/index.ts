export { Input, BackgroundProp } from './Input';
export { ErrorIcon } from './Icons';
