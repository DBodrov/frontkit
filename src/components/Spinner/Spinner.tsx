import classnames from 'classnames';
import React from 'react';

import styles from './Spinner.module.css';

interface SpinnerProps extends React.HTMLAttributes<HTMLElement> {
    /** Class names passed to spinner wrapper to change styling */
    className?: string;
    /** Inline style objects passed to spinner wrapper */
    style?: React.StyleHTMLAttributes<HTMLElement>;
    /** String passed to spinner to change color */
    color?: string;
    /** ID attribute for QA Auto-tests
     * @default Spinner
     * */
    dataTestId?: string;
}

interface CircleProps extends React.HTMLAttributes<HTMLElement> {
    color: string;
}

const circleCls = classnames(styles.circle, styles.lineAnimation);

const Circle = ({ color }: CircleProps) => <div className={circleCls} style={{ backgroundColor: color }} />;

export function Spinner({ className, style, color = '#79b5ff', dataTestId = 'Spinner' }: SpinnerProps): JSX.Element {
    return (
        <div data-testid={dataTestId} className={className} style={style}>
            <div className={styles.spinner}>
                <Circle color={color} />
                <Circle color={color} />
                <Circle color={color} />
                <Circle color={color} />
            </div>
        </div>
    );
}
