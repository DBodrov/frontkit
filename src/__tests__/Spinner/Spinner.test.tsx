import React from 'react';
import { render } from '@testing-library/react';

import { Spinner } from '../../components/Spinner';

describe('<Spinner />', () => {
    const spinnerTestId = 'Spinner';

    test('should spinner rendered', () => {
        const { getByTestId } = render(<Spinner />);
        const spinner = getByTestId(spinnerTestId);
        expect(spinner).toMatchSnapshot();
    });
    test('should have a color', () => {
        const { getByTestId } = render(<Spinner color="#0dd9d5" />);
        const spinner = getByTestId(spinnerTestId);
        expect(spinner.children[0].children[0]).toHaveStyle('background-color: #0dd9d5');
    });
    test('should have a passed className', () => {
        const className = 'Spinnerrrrrrr';
        const { getByTestId } = render(<Spinner dataTestId={spinnerTestId} className={className} />);
        const spinner = getByTestId(spinnerTestId);
        expect(spinner).toHaveClass('Spinnerrrrrrr');
    });
});
