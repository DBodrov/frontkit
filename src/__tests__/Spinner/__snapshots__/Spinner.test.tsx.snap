// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should spinner rendered 1`] = `
<div
  data-testid="Spinner"
>
  <div
    class="spinner"
  >
    <div
      class="circle lineAnimation"
      style="background-color: rgb(121, 181, 255);"
    />
    <div
      class="circle lineAnimation"
      style="background-color: rgb(121, 181, 255);"
    />
    <div
      class="circle lineAnimation"
      style="background-color: rgb(121, 181, 255);"
    />
    <div
      class="circle lineAnimation"
      style="background-color: rgb(121, 181, 255);"
    />
  </div>
</div>
`;
